exchanges/bitmex/bitmex_websocket.go

```diff
@@ -123,201 +124,322 @@ func (b *Bitmex) wsHandleIncomingData() {
+			}
 
-					p := currency.NewPairFromString(orderbooks.Data[0].Symbol)
+			if announcement.Action == bitmexActionInitialData {
+				return nil
+			}
+
+			b.Websocket.DataHandler <- announcement.Data
+		case bitmexWSAffiliate:
+			var response WsAffiliateResponse
+			err = json.Unmarshal(respRaw, &response)
+			if err != nil {
+				return err
+			}
+			b.Websocket.DataHandler <- response
+		case bitmexWSInstrument:
+			// ticker
+		case bitmexWSExecution:
+			// trades of an order
+			var response WsExecutionResponse
+			err = json.Unmarshal(respRaw, &response)
+			if err != nil {
+				return err
+			}
+
+			for i := range response.Data {
+				p := currency.NewPairFromString(response.Data[i].Symbol)
+				var a asset.Item
+				a, err = b.GetPairAssetType(p)
+				if err != nil {
+					return err
+				}
+				var oStatus order.Status
+				oStatus, err = order.StringToOrderStatus(response.Data[i].OrdStatus)
+				if err != nil {
+					return err
+				}
+				var oSide order.Side
+				oSide, err = order.StringToOrderSide(response.Data[i].Side)
+				if err != nil {
+					return err
+				}
+				var ts time.Time
+				ts, err = time.Parse(time.RFC3339, response.Data[i].Timestamp)
+				if err != nil {
+					return err
+				}
+				b.Websocket.DataHandler <- &order.Modify{
+					Exchange:  b.Name,
+					ID:        response.Data[i].OrderID,
+					AccountID: strconv.FormatInt(response.Data[i].Account, 10),
+					AssetType: a,
+					Pair:      p,
+					Status:    oStatus,
+					Trades: []order.TradeHistory{
+						{
+							Price:       response.Data[i].Price,
+							Amount:      response.Data[i].OrderQty,
+							Exchange:    b.Name,
+							TID:         response.Data[i].ExecID,
+							Description: "",
+							Type:        "",
+							Side:        oSide,
+							Timestamp:   ts,
+							IsMaker:     false,
+						},
+					},
+				}
+			}
+		case bitmexWSOrder:
+			var response WsOrderResponse
+			err = json.Unmarshal(respRaw, &response)
+			if err != nil {
+				return err
+			}
+			switch response.Action {
+			case "update", "insert":
+				for x := range response.Data {
+					p := currency.NewPairFromString(response.Data[x].Symbol)
 					var a asset.Item
 					a, err = b.GetPairAssetType(p)
 					if err != nil {
-						b.Websocket.DataHandler <- err
-						continue
+						return err
 					}
-
-					err = b.processOrderbook(orderbooks.Data,
-						orderbooks.Action,
-						p,
-						a)
+					var ts time.Time
+					ts, err = time.Parse(time.RFC3339, response.Data[x].TransactTime)
 					if err != nil {
-						b.Websocket.DataHandler <- err
-						continue
+						return err
 					}
-
-				case bitmexWSTrade:
-					var trades TradeData
-					err = json.Unmarshal(resp.Raw, &trades)
+					var oSide order.Side
+					oSide, err = order.StringToOrderSide(response.Data[x].Side)
 					if err != nil {
-						b.Websocket.DataHandler <- err
-						continue
+						b.Websocket.DataHandler <- errors.New(b.Name + " Unable to convert orderside: " + response.Data[x].Side)
```

Comment:
I would like you're feedback here. The reason I did this is because we're receiving an order which  is important. If for some reason, we haven't coded/the API changes for some status, not storing an order for management because of that seems drastic. I figure logging that we're not up to date with order status/type/side and setting them to their unknown types allows us to keep tracking the order

Reply:
Yeah you make a good point, some do return an error though so its not consistent. @thrasher- @xtda What shall we do here?

Reply:
Order should still be stored, but we need a way to standardise the error so that the calling application can know that a potentially serious field is missing so we can address it ASAP

Reply:
I could create a type which is handled by the routines.go datahandler. Like an `OrderClassificationError` or something to encompass Type, Side and Status. Then a future PR can be around storing the error in the database.

Reply:
I've created `order.ClassificationError` which contains `Exchange`, `OrderID`, and the `Error` from `order.StringToStatus/Type/Side`. It is now handled separately in `routines.go` so if we decide to store these issues in the db, its very simple to do in a future PR 🦀 
There is no more inconsistency with return errs either. It all goes through the datahandler and continues to add the order to the order manager